### PR TITLE
19150 prevent filesystem scm path traversal

### PR DESCRIPTION
**Pull request by ulferts**
https://community.openproject.org/work_packages/19150

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://ulferts-hotfix-prevent-filesystem-scm-path-traversal.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://ulferts-hotfix-prevent-filesystem-scm-path-traversal.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by floriank** [comment]
Wuhu. Failing tests :laughing: @ulferts

File: lib/redmine/scm/base.rb [hunk 1]
```diff
@@ -36,6 +36,14 @@ def all
           @scms
         end
 
+        def configured
+          @scms.select do |scm_name|
+            klass = Kernel.const_get("Repository::#{scm_name}")
```
Comment (ulferts) [inline lib/redmine/scm/base.rb]: Outdated.

Comment (myabc) [inline lib/redmine/scm/base.rb]: @ulferts why not `Repository.const_get(scm_name)`?

**Comment by ulferts** [comment]
@floriank, @Azure7111, @myabc, @NobodysNightmare somebody should take a look at this as soon as possible.

**Comment by floriank** [comment]
:+1: 

I think, you should correct the minor mistake in the doc and follow the suggestion by @myabc

**Pull request merged**